### PR TITLE
Replace the turtle with a more sophisticated one

### PR DESCRIPTION
**Pull request by HendrikRoehm**
The old turtle is replaced with a new one, which is able to scale the drawing to the full image size. Additionally, it draws real arcs using the svg path element. The interface of the turtle was refactored a little bit as well.

### CODE REVIEW

**Review by mschlund — COMMENTED** [review]
can be moved to the "tests" directory -- just learned: pytest also picks up and executes unittest-based files out-of-the box :)

File: src/curves.py [hunk 1]
```diff
@@ -8,190 +8,190 @@
 
 class Curve(ABC):
 
-  lsys : ls.LSystem
-  turtle : mt.Turtle
-  startpoint : str
+    lsys: ls.LSystem
+    turtle: mt.Turtle
+    filename: str
+    postProcessMap: any
```
Comment (mschlund) [inline src/curves.py]: very neat idea to collect such post-process-steps in one member-map! (small hint/TODO: snake-case for everything ... I have to add something to take care of that automatically -- suggestions?)

File: src/my_turtle.py [hunk 2]
```diff
@@ -1,50 +1,181 @@
...
+            "-": Rotation(angle=-angle),
+            "L": Rotation(angle=angle),
+            "R": Rotation(angle=-angle),
+            ")": Arc(angle=angle, rx=stride, ry=stride),
+            "(": Arc(angle=-angle, rx=stride, ry=stride)
+        }
+        super().__init__(movementMap, width=size, height=size, stroke=width)
+
+
+class PathSegment:
```
Comment (mschlund) [inline src/my_turtle.py]: I would propose to move the svg-related classes to a module "svg.py" and let the turtle remain here

File: src/curves.py (hunk 1 above)
Comment (HendrikRoehm) [reply src/curves.py]: Actually, one could do something completely different. All post process steps are actually similar rewriting steps as the main rewritting rules. I would propose to define everything as a pipeline, similar as in tensorflow for example:

```
pipeline = [
    dragoncurverwrite(iterations=4),
    postprocessrewrite(iterations=1),
    postprocessrewrite2(iterations=1)
]
```

**Comment by HendrikRoehm** [comment]
> can be moved to the "tests" directory -- just learned: pytest also picks up and executes unittest-based files out-of-the box :)

I am a fan of having the unit test files next to the code it tests.

**Comment by HendrikRoehm** [comment]
I fixed some problem with the setup.py additionally.

**Comment by mschlund** [comment]
Again: Very, very cool to have a real clean and proper svg-turtle :)
Just two comments:
- We should try to keep PEP8-compliance (maybe apart from the suggested column-size ... I think 80 is maybe too antique -- black relaxes this) as tools we can use autopep8 or black
mainly naming:
    --  functions+variables in snake_case
    -- Classes in PascalCase
- the tests should be in the folder "tests" (there are arguments for both: keeping it close to the code as well as reducing clutter in src -- but in the end it's just about consistency with the current way ^^)

**Comment by HendrikRoehm** [comment]
done.

**Pull request merged**